addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -349,7 +353,9 @@ class MessageList extends Component {
         if (threadCache.isLoaded) {
             const threadCacheMessageIds = threadCache.messages.map(message => message.id);
             if (threadCacheMessageIds.includes(messageId) && this.messageRefFromId(messageId)) {
-                await this._scrollToMessage(messageId);
+                if (this.props.hasScrollAdjust) {
+                    await this._scrollToMessage(messageId);
```

Comment:
I think await should be removed here, hint should be marked as processed as soon as the animation starts.

Reply:
> I think await should be removed here, hint should be marked as processed as soon as the animation starts.

Don't think so, scrolling is not instantaneous (and it could have smooth behaviour), so if you mark it as processed as soon as possible, you may loose adjustment to scroll to last message due to previous last message being visible (which is not, because scrolling hasn't terminated yet).